src/flamenco/runtime/program/fd_config_program.c

```diff
@@ -164,20 +185,33 @@ _process_config_instr( fd_exec_instr_ctx_t ctx ) {
     return FD_EXECUTOR_INSTR_ERR_MISSING_REQUIRED_SIGNATURE;
 
   /* Upgrade to writable handle
-     https://github.com/solana-labs/solana/blob/v1.17.17/programs/config/src/config_processor.rs#L128-L133 */
+     https://github.com/solana-labs/solana/blob/v1.17.17/programs/config/src/config_processor.rs#L128-L129 */
 
   if( FD_UNLIKELY( !fd_borrowed_account_acquire_write( config_acc_rec ) ) )
     return FD_EXECUTOR_INSTR_ERR_ACC_BORROW_FAILED;
 
-  if( FD_UNLIKELY( !fd_instr_acc_is_writable_idx( ctx.instr, 0 ) ) )
-    return FD_EXECUTOR_INSTR_ERR_READONLY_DATA_MODIFIED;
+  /* Upgrade to writable handle
+     https://github.com/solana-labs/solana/blob/v1.17.17/programs/config/src/config_processor.rs#L130-L133 */
 
-  if( FD_UNLIKELY( config_acc_rec->const_meta->dlen < ctx.instr->data_sz ) )
+  if( FD_UNLIKELY( config_acc_rec->const_meta->dlen < ctx.instr->data_sz ) ) {
+    /* TODO Log: "instruction data too large" */
     return FD_EXECUTOR_INSTR_ERR_INVALID_INSTR_DATA;
+  }
+
+  /* Inlined solana_sdk::transaction_context::BorrowedAccount::get_data_mut */
 
-  int modify_err = fd_instr_borrowed_account_modify_idx( &ctx, 0, config_acc_rec->const_meta->dlen, &config_acc_rec );
-  if( FD_UNLIKELY( modify_err!=FD_ACC_MGR_SUCCESS ) )
-    return FD_EXECUTOR_INSTR_ERR_READONLY_DATA_MODIFIED;
+  do {
+    int err;
+    if( !fd_account_can_data_be_changed( ctx.instr, 0, &err ) )
+      return err;
+  } while(0);
```

Comment:
Wondering if we should make these do..while loops a macro? So it's more clear why they are there

Reply:
I was wondering if `ERR_SCOPE` or something would be clearer, but up to you. Don't feel strongly either way :)